include/deal.II/lac/la_parallel_vector.h

```diff
@@ -617,7 +617,14 @@ namespace LinearAlgebra
        * the sum of the square of all entries among all processors).
        */
       virtual real_type l2_norm() const override;
-
+      /**
+       * Return square of the $l_2$-norm.
+       *
+       */
+      real_type norm_sqr () const {
+    	  real_type  l2=this->l2_norm();
```

Comment:
another point -- we always separate definition and declaration

Reply:
It might also make sense for this new function to do the actual computation of the square of the L2 norm, and then `l2_norm()` to call it. I suggest this because ['l2_norm()'](https://github.com/dealii/dealii/blob/master/include/deal.II/lac/la_parallel_vector.templates.h#L1221) already does this and the takes the square root of that number, which this function just squares again.

Reply:
yes, absolutely

Reply:
Indeed -- the `norm_sqr()` function exists to avoid the expensive square root. The right approach is to let `norm_sqr()` do the accumulation that's currently done in `l2_norm()`, and then implement the `l2_norm()` function by just taking the square root of whatever `norm_sqr()` returns. That is, do it the other way around than you are doing here.

Reply:
That was just a temporary fix. I'll  change that.